cyral/resource_cyral_repository.go

```diff
@@ -429,15 +440,25 @@ func resourceRepository() *schema.Resource {
 							Optional:    true,
 						},
 						RepoMongoDBServerTypeKey: {
-							Description:  "Type of the MongoDB server. Allowed values: " + supportedTypesMarkdown(mongoServerTypes()),
+							Description: "Type of the MongoDB server. Allowed values: " + supportedTypesMarkdown(mongoServerTypes()) +
+								".\nThe following conditions apply:\n" +
+								"  - If `" + RepoMongoDBServerTypeKey + "=" + Standalone + "`, then only one `" + RepoNodesKey +
+								"` block can be declared and it must be static. See `" + RepoNodeDynamicKey + "`." +
+								"  - If `" + RepoMongoDBServerTypeKey + "=" + Sharded + "` and the a `" + RepoMongoDBSRVRecordName +
+								"` is provided, then all `" + RepoNodesKey + "` blocks must be declared dynamic, " +
+								"otherwise all of them must be static. See `" + RepoNodeDynamicKey + "`.",
 							Type:         schema.TypeString,
 							Required:     true,
 							ValidateFunc: validation.StringInSlice(mongoServerTypes(), false),
 						},
 						RepoMongoDBSRVRecordName: {
-							Description: "Name of a DNS SRV record which contains cluster topology details",
-							Type:        schema.TypeString,
-							Optional:    true,
+							Description: "Name of a DNS SRV record which contains cluster topology details. The following conditions apply:\n" +
+								"  - If `" + RepoMongoDBSRVRecordName + "` is specified, then all `" + RepoNodesKey + "` blocks can be declared `" +
```

Comment:
```suggestion
								"  - If `" + RepoMongoDBSRVRecordName + "` is specified, then all `" + RepoNodesKey + "` blocks must be declared `" +
```

Reply:
@deepak-cyral , per our previous conversation I understood that all nodes must only be dynamic for `sharded` types. Do you also mean that `replicaset` types must also use `dynamic=true` if the SRV record is provided?

Reply:
Please, see the following conversation: https://github.com/cyralinc/terraform-provider-cyral/pull/361#discussion_r1110482324